### ISSUE TITLE
"not found: value ProvidedJS" happen in 0.15.0

### ISSUE THREAD

**Issue by exoego** [scalacenter/scalajs-bundler#311]
I separate sbt dependencies into `project/Dependencies.scala` like this.

```scala
import sbt._
import Keys._
import org.scalajs.sbtplugin.ScalaJSPlugin._
import org.portablescala.sbtplatformdeps.PlatformDepsPlugin.autoImport._
import autoImport._

object Dependencies {
  val scalatest = Def.setting("org.scalatest" %%% "scalatest" % "3.0.8" % "test")
  val providedjs = Seq(
    ProvidedJS / "handler.js"
  )
}
```

This works for 

* scalajs: 0.6.28
* scalajs-bundler: 0.14.0

But, error occurs when I upgraded `scalajs-bundler` from `0.14.0` to `0.15.0`.

```
[error] /home/vagrant/IdeaProjects/annofab/project/Dependencies.scala:59:7: not found: value ProvidedJS
[error]       ProvidedJS / "handler.js"
[error]       ^
```

**Comment by julienrf** [comment]
Are you now using Scala.js 1.x or still 0.6.28?

**Comment by exoego** [comment]
0.6.28

**Comment by julienrf** [comment]
Maybe something is wrong in the imports and they shadow the `ProvidedJS` member “auto imported” from the `ScalaJSPlugin`. Can you try with the fully qualified name?

~~~ scala
org.scalajs.sbtplugin.ScalaJSPlugin.autoImport.ProvidedJS
~~~

**Comment by exoego** [comment]
Unfortunately, it still happens if `import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport.ProvidedJS` is added, or even if fully-qualified usage.
```
val servers = Seq(
  org.scalajs.sbtplugin.ScalaJSPlugin.autoImport.ProvidedJS / "handler.js"
)
```

**Comment by unarist** [comment]
I've reproduced this issue with `addSbtPlugin("ch.epfl.scala" % "sbt-scalajs-bundler" % "0.15.0")` and it has resolved with `addSbtPlugin("ch.epfl.scala" % "sbt-scalajs-bundler" % "0.15.0-0.6")`.

from [Getting started](https://scalacenter.github.io/scalajs-bundler/getting-started.html):

```
addSbtPlugin("ch.epfl.scala" % "sbt-scalajs-bundler" % "0.15.0")
// Or, for Scala.js 0.6.x
addSbtPlugin("ch.epfl.scala" % "sbt-scalajs-bundler" % "0.15.0-0.6")
```

I think depending `"ch.epfl.scala" % "sbt-scalajs-bundler" % "0.15.0"` indirectly loads ScalaJS 1.x, which removes `ProvidedJS` from the core plugin.

https://www.scala-js.org/doc/project/dependencies.html

> **Scala.js 1.x note:** since Scala.js 1.x, this functionality is not part of the core Scala.js sbt plugin. Instead, it is provided by sbt-jsdependencies. You will need to add `addSbtPlugin("org.scala-js" % "sbt-jsdependencies" % "1.0.0-M1")` to project/plugins.sbt, as well as `enablePlugins(JSDependenciesPlugin)` to your Scala.js projects.

**Comment by exoego** [comment]
Closing since it is just module id issue.

I expect future release should be Scala.js version agnostic (`0.15.1` or later should work both for Scala.js `0.6.x` and `1.0.x`)

**Comment by julienrf** [comment]
> I expect future release should be Scala.js version agnostic (`0.15.1` or later should work both for Scala.js `0.6.x` and `1.0.x`)

This is unfortunately not possible because Scala.js 0.6.x and 1.x are not binary compatible (so, the code of sbt-scalajs-bundler works with either Scala.js 0.6.x or Scala.js 1.x but not both). This is why we have to release two artifacts. But in the future we might switch to a different schema: instead of using two different versions of the plugin (0.15.0 and 0.15.0-0.6), we would use two different names (sbt-scalajs-bundler and sbt-scalajs-bundler-0.6).

**Comment by exoego** [comment]
>  two different names (sbt-scalajs-bundler and sbt-scalajs-bundler-0.6).

That helps my privately-hosted Scala Steward.
Since it currently does not understand `0.15.0` is not for Scala.js 0.6.x.

**Comment by exoego** [comment]
FYI:
It seems that new artifact `sbt-scalajs-bundler-0-6` for Scala.js 0.6.x was published starting from version `0.16.0`.

```sbt
addSbtPlugin("ch.epfl.scala" % "sbt-scalajs-bundler-0-6" % "0.16.0")
```

**Comment by julienrf** [comment]
Hello, actually I've changed the artifact name for 'sbt-scalajs-bundler-sjs06'.
